pandas/core/dtypes/common.py

```diff
@@ -1013,11 +1016,7 @@ def is_datetime64_any_dtype(arr_or_dtype) -> bool:
     False
     >>> is_datetime64_any_dtype(np.array([], dtype=np.datetime64))
     True
-    >>> is_datetime64_any_dtype(pd.DatetimeIndex([1, 2, 3],
-                                dtype=np.datetime64))
```

Comment:
I think you can just change dtype here to `"datetime64[ns]"` - not sure on general rules for passing `np.datetime64` without a base as a dtype argument but cc @jbrockmendel

Reply:
yah, this will raise ValueError.  Either don't pass a dtype at all or pass "datetime64[ns]"